core/src/main/java/org/infinispan/factories/InterceptorChainFactory.java

```diff
@@ -323,11 +289,79 @@ private AsyncInterceptorChain buildInterceptorChain() {
 
       AsyncInterceptor callInterceptor = createInterceptor(new CallInterceptor(), CallInterceptor.class);
       interceptorChain.appendInterceptor(callInterceptor, false);
+
+      if (configuration.persistence().usingStores()) {
+         addPersistenceInterceptors(interceptorChain, configuration, configuration.persistence());
+      }
+
       log.trace("Finished building default interceptor chain.");
       buildCustomInterceptors(interceptorChain, configuration.customInterceptors());
       return interceptorChain;
    }
 
+   public void addPersistenceInterceptors(AsyncInterceptorChain interceptorChain, Configuration cacheConfiguration, PersistenceConfiguration persistenceConfiguration) {
```

Comment:
The issue is, when adding the persistence interceptor at runtime, it is hard to know the specific spot without calculating it, and the logic is inside ```buildInterceptorChain```, so I extracted the logic to be able to reuse it. 
I am open to suggestions to avoid this refactoring, is there a place in the interceptor stack where I can safely insert the persistence interceptors after the fact?

Reply:
I would have preferred to always add the persistence interceptors immediately after `EntryWrappingInterceptor`, even if it changed the order a bit. The advantage is that when a module adds another interceptor before EWI, the index is no longer valid, but you can always look up EWI and add the persistence interceptors after it.

Reply:
That looks good @danberindei I had it initially like this, but was unsure if it'd cause any issue not covered by the testsuite. I will send another PR

Reply:
https://github.com/infinispan/infinispan/pull/9506